com.unity.render-pipelines.universal/Runtime/Debug/DebugDisplaySettingsMaterial.cs

```diff
@@ -0,0 +1,379 @@
+        internal AlbedoDebugValidationPreset albedoDebugValidationPreset
+        {
+            get => m_AlbedoDebugValidationPreset;
+            set
+            {
+                m_AlbedoDebugValidationPreset = value;
+                AlbedoDebugValidationPresetData presetData = m_AlbedoDebugValidationPresetData[(int)value];
+                AlbedoMinLuminance = presetData.minLuminance;
+                AlbedoMaxLuminance = presetData.maxLuminance;
+                AlbedoCompareColor = presetData.color;
+            }
+        }
+
+        internal float AlbedoMinLuminance = 0.01f;
+        internal float AlbedoMaxLuminance = 0.90f;
+
+        float m_AlbedoHueTolerance = 0.104f;
+        internal float AlbedoHueTolerance
+        {
+            get => m_AlbedoDebugValidationPreset == AlbedoDebugValidationPreset.DefaultLuminance ? 1.0f : m_AlbedoHueTolerance;
+            private set => m_AlbedoHueTolerance = value;
+        }
+
+        float m_AlbedoSaturationTolerance = 0.214f;
+        internal float AlbedoSaturationTolerance
+        {
+            get => m_AlbedoDebugValidationPreset == AlbedoDebugValidationPreset.DefaultLuminance ? 1.0f : m_AlbedoSaturationTolerance;
+            private set => m_AlbedoSaturationTolerance = value;
+        }
+
+        internal Color AlbedoCompareColor = new Color(127f / 255f, 127f / 255f, 127f / 255f, 255f / 255f);
+
+        internal float MetallicMinValue = 0.0f;
+        internal float MetallicMaxValue = 0.9f;
+
+        internal DebugMaterialValidationMode MaterialValidationMode;
+
+        #endregion
+
+        internal DebugMaterialMode DebugMaterialModeData { get; private set; }
+        internal DebugVertexAttributeMode DebugVertexAttributeIndexData { get; private set; }
+
+        static void DebugMaterialValidationModeChanged(DebugUI.Field<int> field, int value)
+        {
+            // Hacky way to hide non-relevant UI options based on displayNames.
+            var mode = (DebugMaterialValidationMode)value;
+            var validationWidgets = field.parent.children;
+            foreach (var widget in validationWidgets)
+            {
+                if (mode == DebugMaterialValidationMode.None && (
+                    widget.displayName == k_AlbedoSettingsContainerName ||
+                    widget.displayName == k_MetallicSettingsContainerName))
+                {
+                    widget.isHidden = true;
+                }
+                else if (mode == DebugMaterialValidationMode.Albedo && widget.displayName == k_MetallicSettingsContainerName)
+                {
+                    widget.isHidden = true;
+                }
+                else if (mode == DebugMaterialValidationMode.Metallic && widget.displayName == k_AlbedoSettingsContainerName)
+                {
+                    widget.isHidden = true;
+                }
+                else
+                {
+                    widget.isHidden = false;
+                }
```

Comment:
Can we find a way to extract the mode conditions outside the loop?

Reply:
I'm not sure if that's possible here, as each condition also depends on the loop variable `widget`.